app/Extensions/DatabaseAuthUserProvider.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+namespace App\Extensions;
+
+use App\User;
+use Doctrine\DBAL\Driver\PDOConnection;
+use Doctrine\DBAL\Driver\PDOException;
+use Illuminate\Contracts\Auth\UserProvider;
+use Illuminate\Contracts\Auth\Authenticatable as UserContract;
+use Illuminate\Database\DatabaseManager;
+
+class DatabaseAuthUserProvider implements UserProvider
+{
+    /**
+     * @var DatabaseManager
+     */
+    protected $dbManager;
+
+    public function __construct(DatabaseManager $dbManager)
+    {
+        $this->dbManager = $dbManager;
+    }
+
+    /**
+     * Retrieve a user by their unique identifier.
+     *
+     * @param  mixed $identifier
+     * @return \Illuminate\Contracts\Auth\Authenticatable|null
+     */
+    public function retrieveById($identifier)
+    {
+        if ($identifier) {
+            return new User(['user' => $identifier]);
+        }
+
+        return null;
+    }
+
+    /**
+     * Retrieve a user by their unique identifier and "remember me" token.
+     *
+     * @param  mixed $identifier
+     * @param  string $token
+     * @return \Illuminate\Contracts\Auth\Authenticatable|null
+     */
+    public function retrieveByToken($identifier, $token)
+    {
+        return $this->retrieveById($identifier);
+    }
+
+    /**
+     * Update the "remember me" token for the given user in storage.
+     *
+     * @param  \Illuminate\Contracts\Auth\Authenticatable $user
+     * @param  string $token
+     * @return void
+     */
+    public function updateRememberToken(UserContract $user, $token)
+    {
+        //
```

Comment:
This is used for "remember me" tokens. As we don't currently have a web login it doesn't matter if it's empty. If we move to a web based authentication and wanted "remember me" functionality we would need to implement this. I think it's fine as is for just now.